SU-HSDO/suhumsci
# HSD8-600 Generic 3 column pattern with variants

# READY FOR REVIEW

# Summary
- Created a new Generic pattern with no styles, only layout structures with some variants.
- Added a twig filter to check if the variable has markup. unfortunately the filter has to be in a module and can't live in the theme.

# Needed By (Date)
- 2/27

# Urgency
- medium

# Steps to Test
1. checkout this branch
1. `composer install --prefer-source` for good measure
1. clear cache/sync a site `blt drupal:sync --site=[sitename]`
1. Create a new test view to display any content type you want
1. add some fields. change the view "SHOW" settings from "fields" to "pattern" and choose the "Generic Three Column" pattern.
1. map the fields to the left, middle, and right regions
1. view the output.
1. change the settings of the pattern to use a different variant. like the "Larger right" or something
1. observe the output changes.


# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
